src/server/pfs/server/testing/server_test.go

```diff
@@ -102,6 +102,68 @@ func finfosToPaths(finfos []*pfs.FileInfo) (paths []string) {
 func TestPFS(suite *testing.T) {
 	suite.Parallel()
 
+	suite.Run("InvalidProject", func(t *testing.T) {
+		t.Parallel()
+		env := testpachd.NewRealEnv(t, dockertestenv.NewTestDBConfig(t))
+
+		require.YesError(t, env.PachClient.CreateProject("/repo"))
+
+		require.NoError(t, env.PachClient.CreateProject("lenny"))
+		require.NoError(t, env.PachClient.CreateProject("lenny123"))
+		require.NoError(t, env.PachClient.CreateProject("lenny_123"))
+		require.NoError(t, env.PachClient.CreateProject("lenny-123"))
+
+		require.YesError(t, env.PachClient.CreateProject("lenny.123"))
+		require.YesError(t, env.PachClient.CreateProject("lenny:"))
+		require.YesError(t, env.PachClient.CreateProject("lenny,"))
+		require.YesError(t, env.PachClient.CreateProject("lenny#"))
+	})
+
+	suite.Run("DefaultProject", func(t *testing.T) {
+		t.Parallel()
+		env := testpachd.NewRealEnv(t, dockertestenv.NewTestDBConfig(t))
+
+		// the default project should already exist
+		pp, err := env.PachClient.ListProject()
+		require.NoError(t, err)
+		var sawDefault bool
+		for _, p := range pp {
+			if p.Project.Name == pfs.DefaultProjectName {
+				sawDefault = true
+			}
+		}
+		require.True(t, sawDefault)
+		// this should fail because the default project already exists
+		require.YesError(t, env.PachClient.CreateProject(pfs.DefaultProjectName))
+		// but this should succeed
+		var desc = "A different description."
+		require.NoError(t, env.PachClient.UpdateProject(pfs.DefaultProjectName, desc))
+		// and it should have taken effect
+		pp, err = env.PachClient.ListProject()
+		require.NoError(t, err)
+		for _, p := range pp {
+			if p.Project.Name == pfs.DefaultProjectName {
+				require.Equal(t, desc, p.Description)
+			}
+		}
+		// deleting the default project is allowed, too
+		require.NoError(t, env.PachClient.DeleteProject(pfs.DefaultProjectName, false))
```

Comment:
Loosley related to the test, but doesn't necessarily have to happen in this PR - I don't think we should let the user get into a state where they don't have any projects, and I think that it may make sense to prevent the user from deleting the Default project to achieve those means. Probably worth aligning with Product

Reply:
Agreed. Left a note in the PRD for discussion.